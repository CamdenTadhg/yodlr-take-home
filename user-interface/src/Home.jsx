import React from 'react';
import './Home.css';

const Home = () => {
    return (
        <p className='Home'>Welcome to Yodlr</p>
    );
};

export default Home;