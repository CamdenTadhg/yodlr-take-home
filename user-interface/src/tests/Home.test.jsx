import {render} from '@testing-library/react';
import {it, expect} from 'vitest';
import {MemoryRouter} from 'react-router-dom';
import Home from '../Home';

it('renders without crashing', () => {
    render(
        <MemoryRouter>
            <Home/>
        </MemoryRouter>
    );
});

it('matches the snapshot', () => {
    const {asFragment} = render(
        <MemoryRouter>
            <Home />
        </MemoryRouter>
    );
    expect(asFragment()).toMatchSnapshot();
})

it('displays the correct content', () => {
    const {getByText} = render(
        <MemoryRouter>
            <Home/>
        </MemoryRouter>
    );
    expect(getByText(/Welcome to Yodlr/i)).toBeInTheDocument();
})