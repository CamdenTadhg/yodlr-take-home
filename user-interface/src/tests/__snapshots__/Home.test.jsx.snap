// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`matches the snapshot 1`] = `
<DocumentFragment>
  <p
    class="Home"
  >
    Welcome to Yodlr
  </p>
</DocumentFragment>
`;
